packages/docusaurus-theme-classic/src/index.ts

```diff
@@ -151,22 +154,38 @@ export default function docusaurusThemeClassic(
       return modules;
     },
 
-    configureWebpack() {
+    configureWebpack(config) {
       const prismLanguages = additionalLanguages
         .map((lang) => `prism-${lang}`)
         .join('|');
 
-      return {
-        plugins: [
-          // This allows better optimization by only bundling those components
-          // that the user actually needs, because the modules are dynamically
-          // required and can't be known during compile time.
-          new ContextReplacementPlugin(
-            /prismjs[\\/]components$/,
-            new RegExp(`^./(${prismLanguages})$`),
-          ),
-        ],
-      };
+      // This allows better optimization by only bundling those components
+      // that the user actually needs, because the modules are dynamically
+      // required and can't be known during compile time.
+      config.plugins?.push(
+        new ContextReplacementPlugin(
+          /prismjs[\\/]components$/,
+          new RegExp(`^./(${prismLanguages})$`),
+        ),
+      );
+
+      config.module?.rules?.forEach((rule) => {
+        if (typeof rule !== 'string' && Array.isArray(rule.use)) {
+          rule.use.forEach((useItem) => {
+            if (
+              typeof useItem !== 'string' &&
+              'loader' in useItem &&
+              useItem.loader?.includes('docusaurus-mdx-loader')
+            ) {
+              useItem.options ??= {};
+              (useItem.options as MDXOptions).remarkPlugins ??= [];
+              (useItem.options as MDXOptions).remarkPlugins.push(admonitions);
```

Comment:
We need to support custom admonitions as we previously did. We have users heavily using them.

Reply:
agree 👍 

can you show example sites using custom admonitions and the configs used?

Reply:
Site: https://sonceri.art/docs/intro 
Source: https://github.com/lorikrell/gamerchic/blob/ec6e1595a51b7e65538262dd9308fb7517e42f9f/docusaurus.config.js#L359-L402

Reply:
@Josh-Cena I think we should also study how changes in this PR will look like after MDX 2 and https://github.com/remarkjs/remark-directive

I don't think the current API for admonition customization will stay as is, and instead we'll do customizations through swizzle?

So maybe we can already do the breaking change today if it's already planned anyway 🤷‍♂️